Compile error with bytemuck <1.11 due to use of `zeroed_vec`

### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.


### Reproducible example

Use polars as a dependency in a crate with bytemuck at a lower version than 1.11 for any reason. E.g.:

Run `cargo new` to make an empty project, edit the Cargo.toml file dependencies section to:
```toml
[depdencies]
bytemuck = "<=1.10.0"
polars = "0.35.4"
```



### Log output

```shell
Compiling polars-utils v0.35.4
error[E0432]: unresolved import `bytemuck::allocation::zeroed_vec`
 --> /Users/colin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-utils-0.35.4/src/cache.rs:7:5
  |
7 | use bytemuck::allocation::zeroed_vec;
  |     ^^^^^^^^^^^^^^^^^^^^^^----------
  |     |                     |
  |     |                     help: a similar name exists in the module: `zeroed_box`
  |     no `zeroed_vec` in `allocation`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `polars-utils` (lib) due to previous error
```
```


### Issue description

`polars-utils` and `polars-ops` appear to use `zeroed_vec`, which was added in bytemuck 1.11, but the dependency is only pinned to version 1.

(I think this just needs to pin to version >=1.11.)

### Expected behavior

The package should compile.

### Installed versions

(see above minimal cargo.toml dependency section)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@ atoi = "2"
 avro-schema = { version = "0.3" }
 base64 = "0.21.2"
 bitflags = "2"
-bytemuck = { version = ">=1.11.0,<2", features = ["derive", "extern_crate_alloc"] }
+bytemuck = { version = "1.11", features = ["derive", "extern_crate_alloc"] }
 chrono = { version = "0.4.31", default-features = false, features = ["std"] }
 chrono-tz = "0.8.1"
 ciborium = "0.2"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@ atoi = "2"
 avro-schema = { version = "0.3" }
 base64 = "0.21.2"
 bitflags = "2"
-bytemuck = { version = "1", features = ["derive", "extern_crate_alloc"] }
+bytemuck = { version = ">=1.11.0,<2", features = ["derive", "extern_crate_alloc"] }
 chrono = { version = "0.4.31", default-features = false, features = ["std"] }
 chrono-tz = "0.8.1"
 ciborium = "0.2"

